src/common/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -1420,6 +1422,196 @@ class ngraph::pass::mask_propagation::Transpose : public MatcherPass {
+            int64_t total_lengths = 0;
+            for (int i = 0; i < split_lengths.size(); i++) {
+                if (split_lengths[i] == -1) {
+                    minus_one_length_idx = i;
+                    continue;
+                }
+                total_lengths += split_lengths[i];
+            }
+            if (minus_one_length_idx >= 0 && (*input_mask)[axis].size() > 0) {
+                const auto& input_shape = input.get_partial_shape();
+                if (input_shape[axis].is_dynamic())
+                    return false;
+                auto split_dim = static_cast<uint64_t>(input_shape[axis].get_length());
+                split_lengths[minus_one_length_idx] = split_dim - total_lengths;
+            }
+
+            uint64_t split_start = 0;
+            uint64_t split_end = split_lengths[0];
+            std::vector<Mask::Ptr> output_masks;
+            std::vector<Mask*> output_masks_raw;
+            Mask* input_mask_raw = input_mask.get();
+            for (size_t i = 0; i < split->get_output_size(); i++) {
+                auto output_mask = std::make_shared<Mask>();
+                output_mask->copy_and_slice_mask_from(input_mask_raw, axis, split_start, split_end);
+                output_masks.push_back(output_mask);
+                output_masks_raw.push_back(output_mask.get());
+                output_mask->add_callback(
+                    [input_mask_raw, axis, split_start, split_end](Mask::Ptr cur_mask) -> bool {
+                        cur_mask->copy_and_slice_mask_from(input_mask_raw, axis, split_start, split_end);
+                        return true;
+                    },
+                    input_mask);
+                split_start = split_end;
+                if (i + 1 < split_lengths.size())
+                    split_end += split_lengths[i + 1];
+                setMask(split->output(i), output_masks[i]);
+            }
+
+            auto input_mask_callback = [output_masks_raw, axis, split_lengths](Mask::Ptr cur_mask) -> bool {
+                cur_mask->clean_dim_values();
+                uint64_t split_start = 0;
+                uint64_t split_end = split_lengths[0];
+                for (size_t i = 0; i < output_masks_raw.size(); i++) {
+                    auto mask = output_masks_raw[i];
+                    for (size_t j = 0; j < mask->size(); j++) {
+                        const auto& set = (*mask)[j];
+                        if (j == axis) {
+                            for (auto d : set)
+                                (*cur_mask)[j].insert(d + split_start);
+                        } else if (!set.empty()) {
+                            (*cur_mask)[j] = set;
+                        }
+                    }
+                    split_start = split_end;
+                    if (i + 1 < split_lengths.size())
+                        split_end += split_lengths[i + 1];
+                }
+                cur_mask->initialize_dependencies();
```

Comment:
Are you sure you need `initialize_dependencies` here?

Reply:
It may be redundant

Reply:
Please check if it's redundant, because if it is, it could lead to infinite loop problem. `initialize_dependencies` should be called only in cases you are sure you need to update neigbors

Reply:
I checked that already